test/integration/protocol_integration_test.cc

```diff
@@ -2119,9 +2119,13 @@ name: passthrough-filter
   initialize();
   codec_client_ = makeHttpConnection(lookupPort("http"));
 
-  // Sends a request with headers and data.
-  changeHeadersForStopAllTests(default_request_headers_, false);
-  auto encoder_decoder = codec_client_->startRequest(default_request_headers_);
+  // Sends a POST request with headers and data.
+  Http::TestRequestHeaderMapImpl request_headers{{":method", "POST"},
```

Comment:
why does this test need to be changed?

Reply:
makes sense.  let's just  default_request_headers_.setMethod("POST"); here and in other places to minimize diffs, and call it a day.

Reply:
actually as discussed in person, let's see if we can avoid the  buffering for only-retrying-for-"safe"-http3.  If not, stick with the above comment :-)

Reply:
It works!